Fix some hyperlinks

stantios opened the pull request:
Description: This PR fixes hyperlinks in the README file. I have also found other hyperlinks that are broken but I am not sure where they could be redirected to. One more thing, I have also changed the CurseForge link to remove the '/placeholder' since it was an invalid link. Let me know if you want me to revert that change back since I would be happy to.

The ones that are broken are:

- ForgeDev
- Pull-Requests

I found that these two links did not have a page to redirect to or I could not find a page to redirect them to on the current wiki, so I left those hyperlinks alone.

stantios committed:
Fix some hyperlinks

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/neoforged/NeoForge?pullRequest=202) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/neoforged/NeoForge?pullRequest=202) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: Imzxhir<br/>:x: Zahir<br/><hr/>**Zahir** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/neoforged/NeoForge?pullRequest=202) it.</sub>

Commit:
Fix two more hyperlinks

stantios commented:
For now, I have redirected the broken hyperlinks to the Forge Documentation until we have pages in the NeoForge Documentation for those hyperlinks to redirect to.

Hope you don't mind.

sciwhiz12 commented:
Please don't link to the Minecraft Forge documentation site in the NeoForge repository. I'd rather we remove/comment out the link temporarily rather than redirect it to Forge, which wouldn't fit our new contributing flow due to the new NeoGradle setup and current (and upcoming) changes to the contributing flow.

stantios commented:
Oh ok. That's my bad, sorry. I will revert the changes now, and will most likely remove the links so they can be added back later when we have the pages set up.

Commit:
Remove two hyperlinks

stantios commented:
I have now removed the traces of the Minecraft Forge Documentation and removed parts of the wiki that related to the broken hyperlinks. We can add them back with fixed links when we have setup the related pages for them in the NeoForge docs.

Edit: Forgot to mention another change, but I also fixed a grammar error with the word open collective.

sciwhiz12 commented on docs/README.md:
```diff
@@ -31,17 +31,8 @@ For support and questions, visit [the NeoForged Discord server][Discord].
 
 [See the "Getting Started" section in the NeoForged Documentation][Getting-Started].
 
-# Contribute to Forge
```
This header should be kept, to separate the below sections on pull requests and the CLA from the 'Creating Mods' header. The first paragraph should also be kept, and references to Forge should be renamed to NeoForge.

stantios replied:
This should be fixed in this commit (https://github.com/neoforged/NeoForge/pull/202/commits/46a1f85720a61bac365142e47816399e8987d1ea).

Commit:
Fix some issues and add back the section about contributing to NeoForge

stantios commented:
For some reason, it is saying that I haven't signed the CLA but I clearly have as shown here:

![image](https://github.com/neoforged/NeoForge/assets/98621617/8b9e1e3a-343f-404c-baf8-a054c84cdb8b)

sciwhiz12 commented on docs/README.md:
```diff
@@ -31,38 +31,29 @@ For support and questions, visit [the NeoForged Discord server][Discord].
 [See the "Getting Started" section in the NeoForged Documentation][Getting-Started].
 
-# Contribute to Forge
+### Contribute to NeoForge
```
The level of this header should be equal to the level of the `Creating Mods` header. Either this header should be H1 (`#`), or this and the other headers above should be H2 (`##`).

stantios replied:
I have changed this header and the others to H2 since I feel like it would be a better choice.

sciwhiz12 reviewed (changes requested):
Please rewrite your commits to have the same consistent email, for the CLA checker's sake.

Commit:
Change CLA link to NeoForge and change level of header

Commit:
Change level of headers to H2

Commit:
Fix two more hyperlinks

stantios commented:
> Please rewrite your commits to have the same consistent email, for the CLA checker's sake.

I recently changed my email address so I had to setup commit signing again with the right email, so I think it has something to do with that.

stantios committed:
Merge remote-tracking branch 'origin/patch-1' into patch-1

stantios commented:
I am going to close this PR and start from the beginning, if you don't mind.

sciwhiz12 commented:
> > Please rewrite your commits to have the same consistent email, for the CLA checker's sake.
> 
> I recently changed my email address so I had to setup commit signing again with the right email, so I think it has something to do with that.

Yeah, that's probably it. You'll need to rewrite all your commits in this branch to have your new email address on it. Once you do that on your local branch, force-push your local branch to the PR branch.

Or, alternatively (as you've just done), you can redo the PR from the start. If you do, please link to this PR from that new PR's description.

stantios commented:
Yep, I will.
